### DISCUSSION TITLE
New - Dotcom Footer

### DISCUSSION THREAD

**Discussion by adamdot** [Legal-and-General/canopy#824] [Canopy requests]
**Type of request**
New

**Component, pattern or foundation?**
Component

**Nature of request**
I'm working on a footer for our new dotcom-platform sites and I think we should be adding a new component instead of updating one of the footers that currently exist with new modification options.

All I have for design is this image:
![image-20220311-164028](https://user-images.githubusercontent.com/73697373/165713307-9023c3d1-c425-4f9d-b77b-7cb603dfba68.png)

Although I've spoken with James Hayden-Ellis about the existing footers in Canopy and we've identified the Compact footer to be the closest thing to what we want, I think it would be better to create a new one as our layout is different and we will need to add new options that don't exist on any of the others.

Happy to hear all thoughts and suggestions.

**Comment by DaveL-Landg** [comment]
Agreed Adam - my view is we need to thinking about it as a reusable component to create a standard that could be used in other front-ends

**Reply by james-hayden-ellis** [reply]
Just to clarify that the image is a reference to the current website and depicts content not design.

**Comment by elenagarrone** [comment]
Thanks for opening the request @adamdot 👍 

I can see there are some slight differences in the layout compared to the "compact" version. Main one being the copyrights aligned to the left and the background colour. In general I don't believe there are enough differences to make a new component 🤔 

Could you clarify what are the options that don't exist at the moment? Is it background colour, copyright position and terms?

Is there a real need for the above options?

Could someone provide a screenshot of the actual design please? 

My opinion is that we should always try to reuse what's already been built. Modifications and improvements are always welcome but we should have one component that is the same across applications and websites otherwise we will end up exactly in the situation we are now where users jump between one VI to the other.

As an aside I had in mind for a while to update the current code for both `header` and `footer` during the next grouped breaking changes as they don't follow our usual Canopy API structure. The new structure I'm envisaging should look like:
```
<lg-footer>
  <lg-footer-primary-links>
    <lg-footer-primary-link><a href="#">first link</a></lg-footer-primary-link>
    <lg-footer-primary-link><a href="#">second link</a></lg-footer-primary-link>
  </lg-footer-primary-links>
  
    <lg-footer-secondary-links>
      <lg-footer-secondary-link><a href="#">first link</a></lg-footer-secondary-link>
      <lg-footer-secondary-link><a href="#">second link</a></lg-footer-secondary-link>
    </lg-footer-secondary-links>
    
    <lg-footer-logos><img src="logo1" /></lg-footer-logos>
    
    <lg-footer-copyrights>The copyrights</lg-footer-copyrights>
    
    <lg-footer-terms>If needed some terms here</lg-footer-terms>
</lg-footer>
```
Things like `lg-footer-primary-links`, `lg-footer-secondary-links` and `lg-footer-logos` can be optional.

**Reply by james-hayden-ellis** [reply]
There's no design as yet, we've only reviewed the Canopy component against the existing DOTCOM footer to see if it warrants a variation / additional design. There's still some question as to where the disclaimer will live and what the branding rules are around how / when we use the logo.

**Comment by alexcanning** [comment]
Presumably we're talking about reskinning an existing Episerver component rather than using / creating a new Angular component in the Canopy repo?

**Comment by alexcanning** [comment]
Closing as this has been handled by the Dotcom Canopy project.
